File: 2021/go/day1.go
package main

import (
	"fmt"
	"math"
	"os"
	"strconv"
	"strings"
)

const FilePath = "2021/inputs/day1.txt"

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func main() {
	data, err := os.ReadFile(FilePath)
	check(err)

	splitted_data := strings.Split(strings.TrimSpace(string(data)), "\n")
	last_measurement := math.Inf(1)
	var increased_count int

	for _, v := range splitted_data {
		measurement, err := strconv.ParseFloat(v, 64)
		check(err)

		if measurement > last_measurement {
			increased_count++
		}
		last_measurement = measurement
	}
	fmt.Printf("Number of times the measurement increased is %v\n", increased_count)
}

File: 2021/python/day1.py
import os
import math

FILE_PATH = os.path.join(
    os.path.dirname(os.path.realpath(__file__)), "..", "inputs", "day1.txt"
)

last_measurement = math.inf
increased_count = 0

with open(FILE_PATH) as f:
    measurements = [int(x) for x in f.readlines()]

for measurement in measurements:
    if measurement > last_measurement:
        increased_count += 1
    last_measurement = measurement

print(" Part 1 ".center(50, "-"))
print(f"The number of times the measurement increased is {increased_count}\n")

increased_count = 0

for i in range(3, len(measurements)):
    a, b, c, d = (
        measurements[i - 3],
        measurements[i - 2],
        measurements[i - 1],
        measurements[i],
    )
    if b + c + d > a + b + c:
        increased_count += 1

print(" Part 2 ".center(50, "-"))
print(f"The number of times the measurement increased is {increased_count}")
